sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -196,6 +197,12 @@ int hda_dsp_ctrl_init_chip(struct snd_sof_dev *sdev)
 
 	hda_dsp_ctrl_misc_clock_gating(sdev, false);
 
+	/* clear WAKE_STS if not in reset */
+	gctl = snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR, SOF_HDA_GCTL);
+	if (gctl & SOF_HDA_GCTL_RESET)
```

Comment:
Do we need to test gctl? Can we clear WAKE_STS unconditionally?

Reply:
Thanks @plbossart , I got the reason why `if (gctl & SOF_HDA_GCTL_RESET)` is needed after reading https://github.com/thesofproject/linux/commit/b37a15188eae9d4c49c5bb035e0c8d4058e4d9b3.